examples/end-to-end-testing/flake.nix

```diff
@@ -0,0 +1,85 @@
+{
+  description = "Example E2E testing";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+  outputs = { self, nixpkgs, crane, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+        };
+        inherit (pkgs) lib;
+        craneLib = crane.lib.${system};
+        src = craneLib.cleanCargoSource (craneLib.path ./.);
+
+        workspace = craneLib.buildPackage {
+          inherit src;
+          pname = "example-e2e";
+          version = "0.1";
+          doCheck = false;
+          nativeBuildInputs = lib.optionals pkgs.stdenv.isDarwin [
+            pkgs.libiconv
+            pkgs.darwin.apple_sdk.frameworks.Security
+          ];
+        };
+
+        runE2ETests = pkgs.runCommand "e2e-tests"
+          {
+            nativeBuildInputs = with pkgs; [
+              retry
+              curl
+              geckodriver
+              firefox
+            ];
+          } ''
+
+          wait-for-connection() {
+            timeout 5s \
+              retry --until=success --delay "1" -- \
+                curl -s "$@"
+          }
+
+          ${workspace}/bin/server &
+          wait-for-connection --fail localhost:8000
+
+          # Firefox likes to write to $HOME
+          HOME="$(mktemp -d)"
+          geckodriver &
+          wait-for-connection localhost:4444
+
+          export SSL_CERT_FILE="${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt"
+          ${workspace}/bin/e2e_tests
```

Comment:
**thought (non-blocking):** Oh. Interesting. I think it works, and nothing wrong with it but generally [build a whole workspace for testing and other things, and then run the e2e script](https://github.com/fedimint/fedimint/blob/44faa9a9a1945914b1f6241d0f905a854235c81b/nix/craneBuild.nix#L140).

The advantage is that the test scripts work all the same both from dev env (`./script/run-some-test.sh`) and CI.

In Flakebox I even added specialized commands to make it cleaner.

Reply:
Certainly! I added a comment to highlight this.